jung-api/src/main/java/edu/uci/ics/jung/graph/DelegateCTree.java

```diff
@@ -45,16 +47,18 @@
 
   @Override
   public Optional<N> predecessor(N node) {
+    checkNotNull(node, "node");
     Set<N> predecessors = delegate.predecessors(node);
-    Preconditions.checkState(predecessors.size() <= 1);
+    checkState(predecessors.size() <= 1);
     return predecessors.isEmpty()
-        ? Optional.<N>empty()
+        ? Optional.empty()
         : Optional.of(Iterables.getOnlyElement(predecessors));
   }
 
   @Override
   public int depth(N node) {
-    Preconditions.checkArgument(delegate.nodes().contains(node));
+    checkNotNull(node, "node");
```

Comment:
You've got a bunch of identical checks in the accessors. Consider creating a private method to sanity check the input:

```
private void sanityCheckNode(N node) {
  checkNotNull(node, "node");
  checkArgument(delegate.nodes().contains(node), NODE_NOT_IN_TREE, node);
}
```

That both reduces the amount of code and makes it easier for us to uniformly change our error handling if desired.

Reply:
Good catch! I'll try to remember to address this if I find the time to improve the tests. :)